### PR TITLE
[docs] Added README

### LINKED ISSUE #11: [docs] README
Copy structure of README from another module and adapt it to this module.

We'll be adding more info to the README as we go along.

### PR DESCRIPTION
**Pull request by pandafy**
Added LICENSE and CONTRIBUTING.rst
Minor bug fix in admin for displaying notifications.
Updated travis for using openwisp-utils development version.

Fixes #11

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -7,6 +7,181 @@ OpenWISP Notifications Module
...
+
+Run tests with:
+
+.. code-block:: shell
+
+    ./runtests.py
+
+Generating Notifications
+------------------------
+In order to simplify generation of notifications, a signal has been provided which should be used
```
Comment (nemesisdesign) [inline README.rst:103]: add a blank line

File: README.rst [hunk 2]
```diff
@@ -7,6 +7,181 @@ OpenWISP Notifications Module
 .. image:: https://coveralls.io/repos/github/openwisp/openwisp-notifications/badge.svg?branch=master
    :target: https://coveralls.io/github/openwisp/openwisp-notifications?branch=master
 
-----------------------
+------------
 
-WIP
+Abstracts code for generation and transmission of notifications in OpenWISP 2.
```
Comment (nemesisdesign) [inline README.rst:12]: I'd put some more effort in this first line.

File: README.rst [hunk 3]
```diff
@@ -7,6 +7,181 @@ OpenWISP Notifications Module
...
++-----------------+-----------------------------------------------------------------------------+
+|  email_subject  | Sets subject of email notification to be sent.                              |
+|                 | Defaults to the truncated description.                                      |
++-----------------+-----------------------------------------------------------------------------+
+|       url       | Adds a URL in email as                                                      | 
+|                 | ``For more information see <url>.``                                         |
+|                 | Default to **None** meaning above message would not be added to the email.  |
++-----------------+-----------------------------------------------------------------------------+
+
+
```
Comment (nemesisdesign) [inline README.rst:163]: remove extra blank line

File: README.rst [hunk 4]
```diff
@@ -7,6 +7,181 @@ OpenWISP Notifications Module
...
+   
+    notify.send(actor, recipient, verb, action_object, target, level, description, **kwargs)
+
+.. note::
+    Since ``openwisp-notifications`` uses ``django-notifications`` under the hood, usage of the 
+    ``notify signal`` has been kept unaffected to maintain consistency with ``django-notifications``.
+    You can learn more about accepted parameters from `django-notifications documentation 
+    <https://github.com/django-notifications/django-notifications#generating-notifications>`_. 
+
+
```
Comment (nemesisdesign) [inline README.rst:148]: remove extra blank line

File: .travis.yml [hunk 5]
```diff
@@ -24,13 +24,15 @@ before_install:
...
+  - pip install -U https://github.com/openwisp/openwisp-utils/tarball/master#egg=openwisp_utils[qa]
 
 install:
   - python setup.py -q develop
-
+  
 before-script:
   - >
-    openwisp-utils-qa-checks \
+    openwisp-qa-check \
```
Comment (nemesisdesign) [inline .travis.yml:35]: can you please create a `run-qa-checks` script as in the other repos? Look at openwisp-monitoring.

**Comment by pandafy** [comment]
Yes @NoumbissiValere , thanks for pointing out, I have made that change. :sweat_smile:

File: .travis.yml [hunk 6]
```diff
@@ -24,13 +24,15 @@ before_install:
   - pip install -U pip wheel
   - pip install $DJANGO
   - pip install -U -r requirements-test.txt
+  # temporary: remove when openwisp-utils 0.5.0 is released
+  - pip install -U https://github.com/openwisp/openwisp-utils/tarball/master#egg=openwisp_utils[qa]
 
 install:
   - python setup.py -q develop
-
+
```
Comment (nemesisdesign) [inline .travis.yml:32]: there's some white space here, please clear it out, enable seeing whitespace in your editor please

File: .travis.yml (hunk 5 above)
Comment (nemesisdesign) [reply .travis.yml:35]: remember this :point_up:

File: README.rst (hunk 2 above)
Comment (nemesisdesign) [reply README.rst:12]: Something like "This module provides email and web notifications for OpenWISP, it's used to notify OpenWISP users about meaningful events in their network."

**Comment by pandafy** [comment]
Sorry @nemesisdesign , to fix problem @NoumbissiValere pointed out, I pushed code from my desktop without pulling from repository first. I have already made those changes.  Apologies for convenience.

**Comment by pandafy** [comment]
I have updated contributing guidelines as suggested in mailing list.

**Comment by nemesisdesign** [comment]
I'll create another issue to remind us to add feature list and some screenshots later on in the project.

**Pull request merged**